### ISSUE TITLE
Orchestra: Usage of action_service in subsequent actions fails

### ISSUE THREAD

**Issue by gfrey** [StackStorm/st2#4351]
##### SUMMARY

On ST2 v2.8.1 I start an action of type python-script from an Orchestra workflow. It uses action_service to retrieve values from datastore, but constantly fails with “HTTPError: 401 Client Error: Unauthorized”, but when calling the action directly it works fine. 

##### ISSUE TYPE
Pick one below and delete the rest:
 - Bug Report

##### STACKSTORM VERSION
```
# st2 --version
st2 2.8.1, on Python 2.7.6
```

##### OS / ENVIRONMENT / INSTALL METHOD
```
# uname -a
Linux 05e344e7db30 4.15.0-29-generic #31-Ubuntu SMP Tue Jul 17 15:39:52 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

Install via the oneline script AFAIK.

##### STEPS TO REPRODUCE

```bash
# st2 key set test 123
```

```bash
# cat test_py_action.yaml 
---
  name: test_py_action
  pack: acp_cs
  runner_type: python-script
  description: Test py action calling the action_service
  enabled: true
  entry_point: test_py_action.py

#  cat test_py_action.py   
from os import sys, path
from lib.base import BaseAction

class TestAction(BaseAction):
    def run(self):
        v = self.action_service.get_value('test', local=False)
        self.logger.debug("---> {}".format(v))

# cat test_wf.yaml 
---
name: "test_wf"
pack: "acp_cs"
runner_type: "orchestra"
description: "A test workflow"
enabled: true
entry_point: "workflows/test_wf.yaml"

#  cat workflows/test_wf.yaml 
---
version: '1.0'
description: A test workflow.

tasks:
  test:
    action: acp_cs.test_py_action
```

##### EXPECTED RESULTS
Should return the values from datastore just fine, without authentication issues, like when calling the 

##### ACTUAL RESULTS
```
st2.actions.python.TestAction: DEBUG    Creating new Client object.
st2.actions.python.TestAction: DEBUG    Retrieving value from the datastore (name=test)
st2.actions.python.TestAction: ERROR    Exception retrieving value from datastore (name=test): 401 Client Error: Unauthorized
MESSAGE: Unauthorized - One of Token or API key required. for url: http://127.0.0.1:9101/v1/keys/test?scope=system&decrypt=false
Traceback (most recent call last):
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/services/datastore.py", line 125, in get_value
    kvp = client.keys.get_by_id(id=name, params=params)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2client/models/core.py", line 40, in decorate
    return func(*args, **kwargs)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2client/models/core.py", line 204, in get_by_id
    self.handle_error(response)
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/st2client/models/core.py", line 166, in handle_error
    response.raise_for_status()
  File "/opt/stackstorm/st2/lib/python2.7/site-packages/requests/models.py", line 928, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
HTTPError: 401 Client Error: Unauthorized
MESSAGE: Unauthorized - One of Token or API key required. for url: http://127.0.0.1:9101/v1/keys/test?scope=system&decrypt=false
st2.actions.python.TestAction: DEBUG    ---> None
```

**Comment by LindsayHill** [comment]
We've just released StackStorm 2.9. This includes fixes to Orquesta. Can you please test this with the new version, and report if it is still an issue?

**Comment by gfrey** [comment]
It is still an issue. What changed is that the WebUI doesn't report the subsequent python action any longer.
v2.9:
<img width="707" alt="screen shot 2018-09-26 at 08 29 41" src="https://user-images.githubusercontent.com/519838/46061293-75ae9200-c166-11e8-91ad-3e78e883f909.png">

v2.8
<img width="756" alt="screen shot 2018-09-26 at 08 29 56" src="https://user-images.githubusercontent.com/519838/46061294-75ae9200-c166-11e8-9af7-01683aab727c.png">

**Comment by LindsayHill** [comment]
What does `st2 execution get <ID>` report for the parent execution?

**Comment by bigmstone** [comment]
@gfrey fixed in https://github.com/StackStorm/st2web/pull/608 which will be put into 2.9.1 coming out in short order.

**Comment by cognifloyd** [comment]
@gfrey do you have an st2 install you don't mind hacking to see if the fix described 
 in https://github.com/StackStorm/st2/issues/4414#issuecomment-434581605 (read the caveats about why I didn't turn this into a PR) makes using action_service work as well? I believe it will.

I patched two files in my 2.9.1 install, and then did `st2ctl restart` to make auth token creation work.
/opt/stackstorm/st2/lib/python2.7/site-packages/st2common/services/workflows.py (line 698)

```diff
-                st2_ctx = {'execution_id': wf_ex_db.action_execution}
+                st2_ctx = {
+                   'execution_id': wf_ex_db.action_execution,
+                   'user': wf_ex_db.context['st2']['user'],
+               }
```

/opt/stackstorm/st2/lib/python2.7/site-packages/st2actions/workflows/workflows.py (line 144)

```diff
-                    st2_ctx = {'execution_id': wf_ex_db.action_execution}
+                    st2_ctx = {
+                        'execution_id': wf_ex_db.action_execution,
+                        'user': wf_ex_db.context['st2']['user'],
+                    }
```

**Comment by m4dcoder** [comment]
Fix at https://github.com/StackStorm/st2/pull/4421
